subprojects/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/GroovyJavaJointIncrementalCompilationIntegrationTest.groovy

```diff
@@ -51,32 +74,28 @@ class GroovyJavaJointIncrementalCompilationIntegrationTest extends AbstractJavaG
-        ['G.groovy', 'J.java'] | ['G.groovy.changed', 'J.java']      | 'no source class mapping file found'     | ['G.groovy.changed', 'J.java']      | 'UP-TO-DATE'
-        ['G.groovy', 'J.java'] | ['G.groovy']                        | 'no source class mapping file found'     | ['G.groovy', 'G2.groovy']           | 'Incremental compilation of '
-        ['G.groovy', 'J.java'] | ['J.java']                          | 'no source class mapping file found'     | ['J.java', 'J2.java']               | 'no source class mapping file found'
-        ['G.groovy', 'J.java'] | ['G.groovy', 'J.java', 'J2.java']   | 'no source class mapping file found'     | ['G.groovy', 'G2.groovy']           | 'no source class mapping file found'
-        ['G.groovy', 'J.java'] | ['G.groovy', 'J.java', 'G2.groovy'] | 'no source class mapping file found'     | ['G.groovy', 'J.java', 'G2.groovy'] | 'UP-TO-DATE'
+        initialSet                            | firstChange                                   | firstBuildMessage                                        | secondChange                                  | secondBuildMessage
+        ['G.groovy']                          | ['G.groovy', 'J.java']                        | 'Groovy-Java joint compilation detected'                 | ['G.groovy', 'J.java']                        | 'UP-TO-DATE'
+        ['J.java']                            | ['G.groovy', 'J.java']                        | 'no source class mapping file found'                     | ['G.groovy', 'J.java.changed']                | 'Groovy-Java joint compilation detected'
+        ['G2.groovy', 'J2.java']              | ['G2.groovy', 'J2.java.changed']              | 'Groovy-Java joint compilation detected'                 | ['G2.groovy', 'J2.java.changed']              | 'UP-TO-DATE'
+        ['G3.groovy', 'J3.java']              | ['G3.groovy.changed', 'J3.java']              | 'unable to find source file of class J3'                 | ['G3.groovy.changed', 'J3.java']              | 'UP-TO-DATE'
+        ['G.groovy', 'J.java']                | ['G.groovy']                                  | 'Groovy-Java joint compilation detected'                 | ['G.groovy', 'G4.groovy']                     | 'Incremental compilation of '
+        ['G.groovy', 'J.java']                | ['J.java']                                    | 'UP-TO-DATE'/*None of the classes needs to be compiled*/ | ['J.java', 'J2.java']                         | 'no source class mapping file found'
+        ['G.groovy', 'J.java']                | ['G.groovy', 'J.java', 'G4.groovy']           | 'Incremental compilation of'                             | ['G.groovy', 'J.java', 'G4.groovy']           | 'UP-TO-DATE'
+        ['G4.groovy', 'G5.groovy', 'J4.java'] | ['G4.groovy.changed', 'G5.groovy', 'J4.java'] | 'unable to find source file of class J4'                 | ['G4.groovy.changed', 'G5.groovy', 'J4.java'] | 'UP-TO-DATE'
```

Comment:
This `where` clause is a compact way to track the test cases we want to cover. I find it pretty hard to read, since there is no information what the current case is about. Moreover, the file names do not give any indications about the dependencies between them (e.g. `J3` depends on `G3`). I think if we make one of the two aspects (what is the test about/how are the dependencies between the files) more apparent, then it would be much easier to read.

Reply:
I added scenario description and made file name more readable, better now?